src/components/customModal/ModalAddHomework.js

```diff
@@ -0,0 +1,213 @@
+import React, { useState, useEffect } from "react";
+import edit from "../../assets/icons/edit-pen.svg";
+import del from "../../assets/icons/delete.svg";
+import { useDispatch, useSelector } from "react-redux";
+import { addHomeworkThunk } from "../../store/slices/homeworksSlice";
+import { openCloseModal, saveModalData } from "../../store/slices/contentSlice";
+import shortid from "shortid";
+import { findCurrentStudyYear } from "../../utils/services";
+import moment from "moment/min/moment-with-locales.min";
+
+export const ModalAddHomework = () => {
+  const [error, setError] = useState(false);
+  const [task, setTask] = useState("");
+  const [page, setPage] = useState("");
+  const [note, setNote] = useState("");
+  const [homeworkData, setHomeworkData] = useState([]);
+  const [editHW, setEditHW] = useState(false);
+  const dispatch = useDispatch();
+  const modalData = useSelector((state) => state.content.openModal.modalData);
+  const modify = useSelector((state) => state.content.openModal.modify);
+  const userId = useSelector((state) => state.user.id);
+
+  const addAnotherHW = () => {
+    if (task || page || note) {
+      setHomeworkData([
+        ...homeworkData,
+        { task: task, page: page, notes: note },
+      ]);
+      setTask("");
+      setPage("");
+      setNote("");
+    } else {
+      setError(true);
+    }
+  };
+
+  const toCloseAndRefreshData = () => {
+    setHomeworkData([]);
+    setError(false);
+    setTask("");
+    setPage("");
+    setNote("");
+    if (modify) {
+      dispatch(openCloseModal({ homeWorkModal: false, editDayModal: true }));
+    } else {
+      dispatch(openCloseModal({ homeWorkModal: false }));
+    }
+  };
+
+  const saveHomework = () => {
+    if (!task && !page && !note && !homeworkData.length) {
+      setError(true);
+    } else {
+      let hw;
+      const homeworkId = shortid.generate();
+      if (task || page || note) {
+        hw = [...homeworkData, { task: task, page: page, notes: note }];
+      } else {
+        hw = homeworkData;
+      }
```

Comment:
Some people think that `let` is a bit hard to follow because you never know in how many places down the code it could be reassigned. These people prefer to use `const` whenever it's possible
```suggestion
      const homeworkId = shortid.generate();
      const hw  = (task || page || note) ?
        ? [...homeworkData, { task: task, page: page, notes: note }]
        : homeworkData;
      
```